src/lang-solidity.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * @license
+ * Copyright (C) 2016 Comandante Cheth
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview
+ * Registers a language handler for Solidity.
+ *
+ * From https://solidity.readthedocs.io:
+ *
+ * > Solidity is a contract-oriented, high-level language whose syntax is
+ * > similar to that of JavaScript and it is designed to target the Ethereum
+ * > Virtual Machine.
+ *
+ * @author Comandante Cheth
+ */
+
+
+(function() {
+  // 8, 16, ..., 256
+  var intSizes = '';
+  for (var i = 8; i <= 256; i += 8) {
+    intSizes += i.toString() + '|';
+  }
+  intSizes = intSizes.slice(0, -1);
+  // 0x8, ..., 0x256, 8x8, ..., 8x248, ..., ..., 248x8
+  var fixedSizes = '';
+  for (var i = 0; i <= 248; i += 8) {
+    for (var j = 8; j <= 256 - i; j += 8) {
+      fixedSizes += i.toString() + 'x' + j.toString() + '|';
+    }
+  }
+  fixedSizes = fixedSizes.slice(0, -1);
+  var keywords = 'abstract|anonymous|as|assembly|break|case|catch|constant' +
```

Comment:
There are a few more [here](https://github.com/ethereum/solidity/blob/cfbbd89dafca57e450f6b4433eb258e6a7f52310/libsolidity/analysis/GlobalContext.cpp), e.g., addmod, sha3, etc.

Actually, after a quick diff with [this lexer](https://github.com/ethereum/solidity/blob/d4da4ef35f8118d49bed54f11392036e1d31f7e7/docs/utils/SolidityLexer.py), it appears the following keywords are missing:

addmod
after
assert
block
ecrecover
event
false
keccak256
msg
mulmod
now
null
revert
ripemd160
selfdestruct
sha256
sha3
suicide
super
this
true
tx